tt_metal/api/kernel.hpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+#include "tt_metal/impl/kernels/kernel_types.hpp"
+
+//==================================================
+//                 KERNEL EXECUTION
+//==================================================
+
+namespace tt::tt_metal{
+namespace v1 {
+
+/**
+ * @brief Sets runtime arguments for a kernel.
+ *
+ * @param program The program containing the kernel.
+ * @param kernel KernelHandle representing the kernel ID.
+ * @param core_spec Specifies the cores where the runtime arguments will be set.
+ * @param runtime_args The runtime arguments to be set.
+ */
+void SetRuntimeArgs(
+    const Program program,
+    KernelHandle kernel,
+    const CoreRangeSet &core_spec,
+    const RuntimeArgs &runtime_args);
+
+/**
+ * @brief Sets multiple runtime arguments of a kernel at once.
+ *
+ * @param program The program containing the kernel.
+ * @param kernel KernelHandle representing the kernel ID.
+ * @param core_spec Vector of core coordinates where the runtime arguments will be set.
+ * @param runtime_args The runtime arguments to be set.
+ */
+void SetRuntimeArgs(
+    const Program program,
+    KernelHandle kernel,
+    const std::vector<CoreCoord> &core_spec,
```

Comment:
```suggestion
    std::span<const CoreCoord> core_spec,
```

Reply:
We cannot do std::span because we need to leave metal able to be compiled with only C++17.

Reply:
@eyonland +1. Today this is critical for automotive as c++20 compilers are **currently** not certified for use.

Reply:
We can use `tt::stl::Span` once https://github.com/tenstorrent/tt-metal/pull/13897 is merged